### PR TITLE
Add ConvNeXt-V2 Model

### PR DESCRIPTION
**Pull request by IMvision12**
# What does this PR do?

Adds ConvNeXt-V2 to transformers.
original repo: https://github.com/facebookresearch/ConvNeXt-V2
paper: https://arxiv.org/abs/2301.00808

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

@alaradirik

### CODE REVIEW

**Review by alaradirik — COMMENTED** [review]
Thank you for working on this, it will be a really nice addition to the library! I left a few comments but most issues are easily fixable.

My main comments are:
- The differences between ConvNext and ConvNextV2 are indeed minimal. However, it might make sense given both models are used as backbones for other models. I'm CC'in @NielsRogge and @sgugger to see if it'd make more sense to make changes ConvNext (V1) and it's configuration to optionally load ConvNext V1 or V2.
- It'd be better to keep the same naming pattern as ConvNext and rename all classes and variables of ConvNeXtV2 to ConvNextV2.
- The conversion script compares the outputs of converted models to ConvNext (V1) output logits, and needs to be updated.
- We create one repo per checkpoint released by the original authors, each repo stores the specific model configuration and preprocessing configuration and weights of that specific checkpoint. 
- You can run `make fixup` to check if all CI tests (style, code quality, repo consistency) tests pass before pushing commits and address issues. You can run all tests with `RUN_SLOW=True pytest tests/models/convnextv2/test_modeling_convnextv2.py`

Hope this helps, I can do a more in-depth review and test the uploaded models once the comments are addressed.

**Comment by sgugger** [comment]
It does look like the model code is exactly the same at a first glance (saw everything is copied from ConvNext). If that is the case, yes to re-using the code of ConvNext, but if we need to make modifications in the convnext modeling file, we should add ConvNext V2 as a new model like in the PR.

**Comment by alaradirik** [comment]
> It does look like the model code is exactly the same at a first glance (saw everything is copied from ConvNext). If that is the case, yes to re-using the code of ConvNext, but if we need to make modifications in the convnext modeling file, we should add ConvNext V2 as a new model like in the PR.

Yes, the code is almost the same, but it adds a Global Response Normalization (GRN) module and removes the layer_scale_parameter from the ConvNeXtV2Layer. Makes more sense to add it as a new model then.

CC @IMvision12

File: src/transformers/models/convnextv2/modeling_convnextv2.py [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+        self.gamma = nn.Parameter(torch.zeros(1, 1, 1, dim))
+        self.beta = nn.Parameter(torch.zeros(1, 1, 1, dim))
+
+    def forward(self, x):
+        Gx = torch.norm(x, p=2, dim=(1, 2), keepdim=True)
+        Nx = Gx / (Gx.mean(dim=-1, keepdim=True) + 1e-6)
+        return self.gamma * (x * Nx) + self.beta + x
+
+
+# Copied from transformers.models.convnext.modeling_convnext.ConvNextLayer with ConvNext->ConvNeXtV2
```
[the remark this answers is not in the document]
Comment (alaradirik) [reply src/transformers/models/convnextv2/modeling_convnextv2.py:162]: You can also check if the copied from statements are valid with `make repo-consistency`

**Comment by IMvision12** [comment]
Thanks for the review @alaradirik I will address all comments!

File: docs/source/en/index.mdx [hunk 2]
```diff
@@ -78,6 +78,7 @@ The documentation is organized into five sections:
 1. **[Conditional DETR](model_doc/conditional_detr)** (from Microsoft Research Asia) released with the paper [Conditional DETR for Fast Training Convergence](https://arxiv.org/abs/2108.06152) by Depu Meng, Xiaokang Chen, Zejia Fan, Gang Zeng, Houqiang Li, Yuhui Yuan, Lei Sun, Jingdong Wang.
 1. **[ConvBERT](model_doc/convbert)** (from YituTech) released with the paper [ConvBERT: Improving BERT with Span-based Dynamic Convolution](https://arxiv.org/abs/2008.02496) by Zihang Jiang, Weihao Yu, Daquan Zhou, Yunpeng Chen, Jiashi Feng, Shuicheng Yan.
 1. **[ConvNeXT](model_doc/convnext)** (from Facebook AI) released with the paper [A ConvNet for the 2020s](https://arxiv.org/abs/2201.03545) by Zhuang Liu, Hanzi Mao, Chao-Yuan Wu, Christoph Feichtenhofer, Trevor Darrell, Saining Xie.
+1. **[ConvNeXtV2](model_doc/convnextv2)** (from <FILL INSTITUTION>) released with the paper [<FILL PAPER TITLE>](<FILL ARKIV LINK>) by <FILL AUTHORS>.
```
[the remark this answers is not in the document]
Comment (IMvision12) [reply docs/source/en/index.mdx:81]: yeah!

File: src/transformers/models/convnextv2/modeling_convnextv2.py [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+class GRN(nn.Module):
+    """GRN (Global Response Normalization) layer"""
+
+    def __init__(self, dim):
+        super().__init__()
+        self.gamma = nn.Parameter(torch.zeros(1, 1, 1, dim))
+        self.beta = nn.Parameter(torch.zeros(1, 1, 1, dim))
+
+    def forward(self, x):
+        Gx = torch.norm(x, p=2, dim=(1, 2), keepdim=True)
```
[the remark this answers is not in the document]
Comment (IMvision12) [reply src/transformers/models/convnextv2/modeling_convnextv2.py:157]: I have added comments explaining what Gx and Nx are will it be fine?
